### ISSUE TITLE
pgmc-supports-no-pie deprecated

### ISSUE THREAD

**Issue by guibou** [tweag/rules_haskell#1803]
**Describe the bug**

rules_haskell, when building with GHC 9.4, spam the output with:

```
on the commandline: warning: [-Wdeprecated-flags]
    -pgmc-supports-no-pie is deprecated: use -pgml-supports-no-pie instead
```

**To Reproduce**

Build with GHC 9.4

**Expected behavior**

It should not spam with this warning.

**Environment**
 - OS name + version: nixos
 - Bazel version: 4.2.2
 - Version of the rules: d18033b7eb3429a55dc4a579b5c19af57ab25e5f

**Additional context**
